### PR TITLE
[UM-4]-creating student class

### PR DESCRIPTION
**Pull request by gorkemiskeceli**

### CODE REVIEW

File: src/main/java/com/archisacadeny/student/Student.java [hunk 1]
```diff
@@ -1,4 +1,49 @@
 package com.archisacadeny.student;
 
+
 public class Student {
+    private long id;
+    private String fullName;
+    private String email;
+    private String password;
+    private String enrolledCourse;
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/student/Student.java]:
```suggestion
    private List<Course> enrolledCourse;
```

File: src/main/java/com/archisacadeny/student/StudentRepository.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package com.archisacadeny.student;
+
+import com.archisacadeny.config.DataBaseConnectorConfig;
+
+import java.sql.SQLException;
+import java.sql.Statement;
+
+public class StudentRepository {
+    public static void sutendtTable(){
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/student/StudentRepository.java]: createStudentTable()

File: src/main/java/com/archisacadeny/student/Student.java [hunk 3]
```diff
@@ -1,4 +1,80 @@
...
+
 public class Student {
+    private long id;
+    private String fullName;
+    private String email;
+    private String password;
+    private String department;
+    private int studentNumber;
+    private String gender;
+    private List<String> enrolledCourse;
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/student/Student.java]:
```suggestion
    private List<Course> enrolledCourse;
```

File: src/main/java/com/archisacadeny/student/StudentRepository.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import java.sql.Statement;
+
+public class StudentRepository {
+    public static void createSutendtTable(){
+
+        try(Statement statement = DataBaseConnectorConfig.getConnection().createStatement())
+        {
+            String query = "DROP SEQUENCE IF EXITS student_id_seq;"+
+                    "CREATE SEQUENCE student_id_seq INCREMENT BY 1 MINVALUE 0 MAXVALUE 2147483647 START 1;"+
+                    "CREATE TABLE IF NOT EXITS students" +
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/student/StudentRepository.java]: there is a syntax error you need to add () after table name

**Review by JamshidS — CHANGES_REQUESTED** [review]

File: src/main/java/com/archisacadeny/student/StudentRepository.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            String query = "DROP SEQUENCE IF EXITS student_id_seq;"+
+                    "CREATE SEQUENCE student_id_seq INCREMENT BY 1 MINVALUE 0 MAXVALUE 2147483647 START 1;"+
+                    "CREATE TABLE IF NOT EXITS students(" +
+                    "id INTEGER DEFAULT maxvalue('student_id_seq') PRIMARY KEY," +
+                    "full_name VARCHAR(255)," +
+                    "email VARCHAR(255)," +
+                    "gender VARCHAR(255),"+
+                    "password VARCHAR(255)," +
+                    "student_number INTEGER,"+
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/student/StudentRepository.java]:
```suggestion
                    "student_number VARCHAR(255),"+
```

File: src/main/java/com/archisacadeny/student/Student.java [hunk 6]
```diff
@@ -1,4 +1,82 @@
...
+
+import java.util.List;
+
 public class Student {
+    private long id;
+    private String fullName;
+    private String email;
+    private String password;
+    private String department;
+    private int studentNumber;
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/student/Student.java]:
```suggestion
    private String studentNumber;
```

File: src/main/java/com/archisacadeny/student/Student.java [hunk 7]
```diff
@@ -1,4 +1,82 @@
 package com.archisacadeny.student;
 
+
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/student/Student.java:3]: and please do share a SS that should show  that your code is working.

**Review by JamshidS — CHANGES_REQUESTED** [review]

File: src/main/java/com/archisacadeny/student/Student.java (hunk 6 above)
Comment (HKalatas) [reply src/main/java/com/archisacadeny/student/Student.java]: Student class ı içinde courseid de olmalı. courseId foreign Key. List<courses>

File: src/main/java/com/archisacadeny/student/StudentRepository.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+                    "CREATE TABLE IF NOT EXITS students(" +
+                    "id INTEGER DEFAULT maxvalue('student_id_seq') PRIMARY KEY," +
+                    "full_name VARCHAR(255)," +
+                    "email VARCHAR(255)," +
+                    "gender VARCHAR(255),"+
+                    "password VARCHAR(255)," +
+                    "student_number VARCHAR(255),"+
+                    "department VARCHAR(255),"+
+                    "gender VARCHAR(255),"+
+                    "enrolled_course VARCHAR(255))";
```
Comment (JamshidS) [inline src/main/java/com/archisacadeny/student/StudentRepository.java:24]: we don't need enrolled_courses field in table

**Review by JamshidS — CHANGES_REQUESTED** [review]

**Pull request closed without merging**